account_financial_report/report/journal_ledger.py

```diff
@@ -0,0 +1,359 @@
+            tax_balance = ml.tax_line_id and ml.balance or 0.0
+        return {
+            "move_line_id": ml.id,
+            "move_id": ml.move_id.id,
+            "date": ml.date,
+            "journal_id": ml.journal_id.id,
+            "account_id": ml.account_id.id,
+            "partner_id": ml.partner_id.id,
+            "label": ml.name,
+            "debit": ml.debit,
+            "credit": ml.credit,
+            "company_currency_id": ml.company_currency_id.id,
+            "amount_currency": ml.amount_currency,
+            "currency_id": ml.currency_id.id,
+            "tax_line_id": ml.tax_line_id.id,
+            "tax_ids": list(ml_taxes.keys()),
+            "base_debit": base_debit,
+            "base_credit": base_credit,
+            "tax_debit": tax_debit,
+            "tax_credit": tax_credit,
+            "base_balance": base_balance,
+            "tax_balance": tax_balance,
+        }
+
+    def _get_account_data(self, accounts):
+        data = {}
+        for account in accounts:
+            data[account.id] = self._get_account_id_data(account)
+        return data
+
+    def _get_account_id_data(self, account):
+        return {
+            "name": account.name,
+            "code": account.code,
+            "internal_type": account.internal_type,
+        }
+
+    def _get_partner_data(self, partners):
+        data = {}
+        for partner in partners:
+            data[partner.id] = self._get_partner_id_data(partner)
+        return data
+
+    def _get_partner_id_data(self, partner):
+        return {"name": partner.name}
+
+    def _get_currency_data(self, currencies):
+        data = {}
+        for currency in currencies:
+            data[currency.id] = self._get_currency_id_data(currency)
+        return data
+
+    def _get_currency_id_data(self, currency):
+        return {"name": currency.name}
+
+    def _get_tax_line_data(self, taxes):
+        data = {}
+        for tax in taxes:
+            data[tax.id] = self._get_tax_line_id_data(tax)
+        return data
+
+    def _get_tax_line_id_data(self, tax):
+        return {"name": tax.name, "description": tax.description}
+
+    def _get_query_taxes(self):
+        return """
+            SELECT aml_at_rel.account_move_line_id, aml_at_rel.account_tax_id,
+            at.description, at.name
+            FROM account_move_line_account_tax_rel AS aml_at_rel
+            LEFT JOIN
+                account_tax AS at on (at.id = aml_at_rel.account_tax_id)
+            WHERE account_move_line_id IN %(move_line_ids)s
+        """
+
+    def _get_query_taxes_params(self, move_lines):
+        return {"move_line_ids": tuple(move_lines.ids)}
+
+    def _get_move_lines(self, move_ids, wizard, journal_ids):
+        move_lines = self.env["account.move.line"].search(
+            self._get_move_lines_domain(move_ids, wizard, journal_ids),
+            order=self._get_move_lines_order(move_ids, wizard, journal_ids),
+        )
```

Comment:
If move_lines is empty query raises an error:

Back
500: Internal Server Error
Traceback
Traceback (most recent call last):
  File "/home/hveficent/Eficent/odoo/13.0/odoo/addons/base/models/ir_http.py", line 229, in _dispatch
    result = request.dispatch()
  File "/home/hveficent/Eficent/odoo/13.0/odoo/http.py", line 804, in dispatch
    r = self._call_function(**self.params)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/http.py", line 345, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/service/model.py", line 93, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/http.py", line 338, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/http.py", line 910, in __call__
    return self.method(*args, **kw)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/http.py", line 510, in response_wrap
    response = f(*args, **kw)
  File "/home/hveficent/Eficent/odoo/addons/addons-13.0/OCA/reporting-engine/report_xlsx/controllers/main.py", line 49, in report_routes
    reportname, docids, converter, **data
  File "/home/hveficent/Eficent/odoo/13.0/odoo/http.py", line 510, in response_wrap
    response = f(*args, **kw)
  File "/home/hveficent/Eficent/odoo/13.0/addons/web/controllers/main.py", line 1956, in report_routes
    html = report.with_context(context).render_qweb_html(docids, data=data)[0]
  File "/home/hveficent/Eficent/odoo/13.0/odoo/addons/base/models/ir_actions_report.py", line 757, in render_qweb_html
    data = self._get_rendering_context(docids, data)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/addons/base/models/ir_actions_report.py", line 774, in _get_rendering_context
    data.update(report_model._get_report_values(docids, data=data))
  File "/home/hveficent/Eficent/odoo/addons/addons-13.0/OCA/account-financial-reporting/account_financial_report/report/journal_ledger.py", line 305, in _get_report_values
    move_lines = self._get_move_lines(move_ids, wizard, journal_ids)
  File "/home/hveficent/Eficent/odoo/addons/addons-13.0/OCA/account-financial-reporting/account_financial_report/report/journal_ledger.py", line 184, in _get_move_lines
    self.env.cr.execute(query_taxes, query_taxes_params)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/sql_db.py", line 163, in wrapper
    return f(self, *args, **kwargs)
  File "/home/hveficent/Eficent/odoo/13.0/odoo/sql_db.py", line 240, in execute
    res = self._obj.execute(query, params)
psycopg2.ProgrammingError: syntax error at or near ")"
LINE 7:             WHERE account_move_line_id IN ()
                                                   ^

Reply:
Thanks @HviorForgeFlow , I see the problem. I add a conditional to validate that case.